packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+
+import {
+  default as ConfirmationModal,
+  Variant as ConfirmationModalVariant,
+} from '@leafygreen-ui/confirmation-modal';
+
+export { ConfirmationModalVariant };
+
+type ConfirmationProperties = {
+  title: string;
+  description: string;
+  buttonText?: string;
+  variant?: ConfirmationModalVariant;
+};
+type ConfirmationCallback = (value: boolean) => void;
+
+interface ConfirmationModalContextData {
+  showConfirmation: (
+    props: ConfirmationProperties,
+    callback: ConfirmationCallback
+  ) => void;
+  isMounted: boolean;
+}
+
+const ConfirmationModalContext =
+  React.createContext<ConfirmationModalContextData>({
+    isMounted: false,
+  } as ConfirmationModalContextData);
+
+class ShowConfirmationEvent extends Event {
+  constructor(
+    public props: ConfirmationProperties,
+    public callback: ConfirmationCallback
+  ) {
+    super('show-confirmation');
+  }
+}
```

Comment:
A more common way of doing custom events in the browser is to use CustomEvents for this, you can also use them to make a more type-safe implementation here, removing the need for a bunch of assertions you are doing inside the component when subscribing to events. A rough idea of how to do this can look like this:

```ts
interface CustomTarget<T> extends EventTarget {
  addEventListener(
    type: 'show-modal',
    callback: ((event: CustomEvent<T>) => void) | null
  ): void;
  addEventListener(
    type: string,
    callback: EventListenerOrEventListenerObject | null
  ): void;
  dispatchEvent(evt: CustomEvent<T>): boolean;
}

type ShowModalEventDetal = {
  title: string;
  // ...
};

const eventTarget: CustomTarget<ShowModalEventDetal> = new EventTarget();

const showConfirmationModal = () => {
  eventTarget.dispatchEvent(
    new CustomEvent('show-modal', { detail: { title: '...' } })
                                            // ^^^ TS will not allow to pass incorrect value here
  );
};

eventTarget.addEventListener('show-modal', (evt) => {
  evt.detail.title;
           // ^^^ TS will correctly provide a value here
});
```